### ISSUE TITLE
Spatial expression prediction for TCGA slides

### ISSUE THREAD

**Issue by kristaxying** [gevaertlab/sequoia-pub#15]
H i～ I want to get the spatial expression prediction for TCGA slides, but when I found the run_visualization.sh, I'm confused with the following code:
############################## get model
    checkpoint = f'{args.model_type}_{args.feat_type}/{study}/'
    obj = read_pickle(checkpoint + 'test_results.pkl')[0]
    gene_ids = obj['genes']

How can I get the needed model or where can I download the pretrained model?

**Comment by mpizurica** [comment]
Hi!

This code is indeed only relevant if you trained your own model locally. To run with a pretrained model, please remove those lines (126-128) and also lines 241-261.

Instead of reading the `gene_ids` variable from `obj`, you can load them instead from the `examples/gene_list.csv` file: 

```
gene_ids = pd.read_csv('gene_list.csv')['gene'].values
```

Then, to load the pretrained SEQUOIA model: 

```
cancer = 'brca'
i = 0 ## fold number
model = ViS.from_pretrained(f"gevaertlab/sequoia-{cancer}-{i}")
```

Note that this loads the model from HuggingFace, so if you haven't already, you will need to log in to your account (see our README instructions in section "Running evaluation of our pre-trained model on an independent dataset").

Hope that is clear, let me know if you have any other questions!

**Comment by kristaxying** [comment]
Thank you for your help. I have made the corresponding changes, but the following code still confuses me. My current input only has one TCGA SVS image, so how should I obtain the required mask?

if 'TCGA' in wsi_file_name:
        slide_path = './TCGA/'+project+'/'
        mask_path = './TCGA/'+project+'_Masks/'
        mask = np.load(mask_path+wsi_file_name.replace('.svs', '')+'/'+'mask.npy')
        manual_resize = None # nr of um/px can be read from slide properties

**Comment by kristaxying** [comment]
I changed the file structures to make sure the import part can work normally.
But when I run the following script

python3 sequoia-pub/src/visualize.py \
                            --study lusc \
                            --project TCGA_LUSE \
                            --wsi_file_name TCGA-37-4132.svs \
                            --gene_names $path/gene_ids/top_1000_gbm.npy \
                            --save_folder top_1000_gbm \
                            --model_type vit \
                            --feat_type vis

I am confused with the following code:
feat_model = timm.create_model("vit_large_patch16_224", img_size=224, 
                                        patch_size=16, init_values=1e-5, 
                                        num_classes=0, dynamic_img_size=True)
        local_dir = "./Gen-Pred/src/spatial_vis/uni_ckpt/"
        feat_model.load_state_dict(torch.load(os.path.join(local_dir, 
                                    "pytorch_model.bin"), map_location=device), strict=True)
        transforms_ = transforms.Compose([
                transforms.Resize(224),
                transforms.ToTensor(),
                transforms.Normalize(mean=(0.485, 0.456, 0.406), std=(0.229, 0.224, 0.225)),
        ])
        feat_model = feat_model.to(device)
        feat_model.eval()

How can I get the "pytorch_model.bin"?

Thanks for your help.
